src/routes/newsRoutes.js

```diff
@@ -0,0 +1,47 @@
+const express = require("express");
+const bodyParser = require("body-parser");
+const newsRoutes = express.Router();
+const verifyToken = require("../middleware/authJWT");
+newsRoutes.use(bodyParser.urlencoded({ extended: false }));
+newsRoutes.use(bodyParser.json());
+const { default: axios } = require("axios");
+const e = require("express");
+const newsAPI ='https://newsapi.org/v2/top-headlines?apiKey=5414cc3f8f0a4615a031eb55478cf173';
+
+//To get news details for selected preference
+newsRoutes.post('/',verifyToken,async(req,res)=>{
+    if (!req.user && req.message == null) {
```

Comment:
can move all checks related to user authentication inside the auth middleware itself.